packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -228,7 +245,7 @@ class BlockListBlock extends Component {
 						style={ this.applyBlockStyle() }
 					>
 						{ isValid ? this.getBlockForType() : <BlockInvalidWarning blockTitle={ title } icon={ icon } /> }
-						{ isSelected && <BlockMobileToolbar clientId={ clientId } /> }
+						<View style={ this.applyToolbarStyle() } >{ isSelected && <BlockMobileToolbar clientId={ clientId } /> }</View>
```

Comment:
I think I'd rather avoid using negative margins here. It seems that we should instead wrap the block content in another view and add the padding to that instead.

Reply:
Actually I was afraid of negative margins as well, but decided to go with them because touching other styles were more dangerous for me.